### PR TITLE
Fixes for installing on nodes imaged by snaps-boot behind a proxy

### LINKED ISSUE #245: Deployments where nodes have multiple NICs have encountered certificate issues.
This was found when creating the snaps-hyperbuild CI scripts. We were able to get around it by not configuring other network interfaces.

The error was:

fatal: [localhost]: FAILED! => {"changed": true, "cmd": ["kubectl", "--kubeconfig=/tmp/snaps-k8s-projects/snaps-hb-ci-boq/node-kubeconfig.yaml", "label", "nodes", "node-1", "zone=node-1"], "delta": "0:00:00.108842", "end": "2019-08-20 16:41:01.863203", "msg": "non-zero return code", "rc": 1, "start": "2019-08-20 16:41:01.754361", "stderr": "Unable to connect to the server: x509: certificate is valid for 10.241.0.1, 10.0.3.11, 10.0.3.11, 10.241.0.1, 127.0.0.1, 10.0.3.11, not 10.0.1.11", "stderr_lines": ["Unable to connect to the server: x509: certificate is valid for 10.241.0.1, 10.0.3.11, 10.0.3.11, 10.241.0.1, 127.0.0.1, 10.0.3.11, not 10.0.1.11"], "stdout": "", "stdout_lines": []}

Guess either the certificate needs to be generated for the 10.0.1 network, or we use 10.0.3 network in k8s config.
also see https://github.com/cablelabs/openhw-hyperbuild/issues/243

### PR DESCRIPTION
**Pull request by spisarski**
#### What does this PR do?
Fixes #245 Sort of
It begins to address the problems outlined in #245 but is really meant to address issues elsewhere driving from installation of snaps-kubernetes on nodes imaged by snaps-boot behind an http proxy
#### Do you have any concerns with this PR?
It still does not fully address why nodes with 2 configured EC2 network interfaces continually fails to deploy properly; however, the existing use cases and the ones required by this fix work fine.
The kubernetes.api_server optional config is really unclear as the value required by k8s appears to be the IP of the last NIC which is really strange but I'm going with it for now.
#### How can the reviewer verify this PR?
run ci
#### Questions:
- Have you connected this PR to the issue it resolves?
yes, but there should be an addendum prior to merging
- Does the documentation need an update?
most likely but there is some for the new optional config attribute
- Does this add new Python dependencies?
no
- Have you added unit or functional tests for this PR?
no but I have expanded the CI environment so we can being triaging the issues with multiple configured EC2 nics.
- Does this patch update any configuration files?
kubernetes.api_server optional config was added

### CODE REVIEW

File: ci/aws/aws-insts.tf [hunk 1]
```diff
@@ -70,3 +75,31 @@ resource "aws_instance" "k8s-node" {
     private_key = file(var.private_key_file)
   }
 }
+
+resource "random_integer" "snaps-k8s-secondary-subnet-mid" {
```
Comment (bo-quan) [inline ci/aws/aws-insts.tf]: This range can be larger (min: 0, max: 255)

File: ci/aws/provisioner.tf [hunk 2]
```diff
@@ -30,16 +33,33 @@ ${var.ANSIBLE_CMD} -u ${var.sudo_user} \
...
   }
 }
 
+# Call ansible script to setup K8s nodes
+resource "null_resource" "snaps-k8s-node-nic-config" {
+  depends_on = [null_resource.snaps-k8s-node-setup]
+
+  # Configure eth1 to secondary subnet
+  provisioner "local-exec" {
+// TODO - swap commands to disable the secondary NIC as it causes problems
```
Comment (bo-quan) [inline ci/aws/provisioner.tf]: remove the commented out lines in this module?

File: ci/playbooks/config_secondary_nic_dhcp.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+  become: yes
+
+  gather_facts: no
+
+  tasks:
+    - name: Add eth1 DHCP config 50-cloud-init.yaml
+      copy:
+        dest: /etc/netplan/50-cloud-init.yaml
+        content: |2
```
Comment (bo-quan) [inline ci/playbooks/config_secondary_nic_dhcp.yaml]: why "|2" (2 at the end)?

File: doc/source/install/install.md [hunk 4]
```diff
@@ -129,6 +129,7 @@ iaas_launch.py. Please see configuration parameters descriptions below.
 | log_level | N | Log level(fatal/error/warn/info/debug/trace) |
 | logging_port | N | Logging Port (e.g. 30011) |
 | node_user | N | The nodes' passwordless sudo user (default: root) |
+| api_host | N | The IP of a master host IP value of the last NIC as that is the one where kubespray places it's certificate (Default: first master host node's IP value)|
```
Comment (bo-quan) [inline doc/source/install/install.md]: "places its"

File: ci/aws/aws-insts.tf [hunk 5]
```diff
@@ -41,6 +41,11 @@ resource "aws_instance" "k8s-build" {
   }
 }
 
+resource "random_integer" "snaps-k8s-primary-subnet-mid" {
```
[the remark this answers is not in the document]
Comment (spisarski) [reply ci/aws/aws-insts.tf]: It is not currently but is in place for when we do fix the multi-nic issue here

File: ci/aws/aws-insts.tf (hunk 1 above)
Comment (spisarski) [reply ci/aws/aws-insts.tf]: We can deal with this later as the second subnet's NICs aren't currently being attached to the VMs

File: ci/aws/provisioner.tf (hunk 2 above)
Comment (spisarski) [reply ci/aws/provisioner.tf]: I left them there on purpose for the next person who works on the NIC issue

File: ci/playbooks/config_secondary_nic_dhcp.yaml (hunk 3 above)
Comment (spisarski) [reply ci/playbooks/config_secondary_nic_dhcp.yaml]: It worked for indenting, before that it was breaking YAML indenting rules. You can remove if it is not necessary...

File: doc/source/install/install.md (hunk 4 above)
Comment (spisarski) [reply doc/source/install/install.md]: I believe it's is correct as it is possessive but my grammar isn't the best :)

**Pull request merged**